src/cron-job.ts

```diff
@@ -0,0 +1,227 @@
+   *
+   * @default false
+   */
+  readonly suspend?: boolean;
+
+  /**
+   * Specifies the number of successful jobs history retained. This would retain the Job and the associated Pod resource and can be useful for debugging.
+   *
+   * @default 3
+   */
+  readonly successfulJobsRetained?: number;
+
+  /**
+   * Specifies the number of failed jobs history retained. This would retain the Job and the associated Pod resource and can be useful for debugging.
+   *
+   * @default 1
+   */
+  readonly failedJobsRetained?: number;
+}
+
+/**
+ * A CronJob is responsible for creating a Job and scheduling it based on provided cron schedule. This helps running Jobs in a recurring manner.
+ */
+export class CronJob extends workload.Workload {
+  /**
+   * The metadata of the recurring `Job` that this cronjob will schedule.
+   */
+  public readonly jobMetadata?: ApiObjectMetadata;
+
+  /**
+   * The schedule this cron job is scheduled to run in.
+   */
+  public readonly schedule: Cron;
+
+  /**
+   * The timezone which this cron job would follow to schedule jobs.
+   */
+  public readonly timeZone?: string;
+
+  /**
+   * The policy used by this cron job to determine the concurrency mode in which to schedule jobs.
+   */
+  public readonly concurrencyPolicy: string;
+
+  /**
+   * The time by which the running cron job needs to schedule the next job execution. The job is considered as failed if it misses this deadline.
+   */
+  public readonly startingDeadline: Duration;
+
+  /**
+   * Whether or not the cron job is currently suspended or not.
+   */
+  public readonly suspend: boolean;
+
+  /**
+   * The number of successful jobs retained by this cron job.
+   */
+  public readonly successfulJobsRetained: number;
+
+  /**
+   * The number of failed jobs retained by this cron job.
+   */
+  public readonly failedJobsRetained: number;
+
+  /**
+   * The properties of the recurring `Job` that this cronjob will schedule.
+   */
+  private readonly jobProperties: JobProps;
+
+  /**
+   * @see base.Resource.apiObject
+   */
+  protected readonly apiObject: ApiObject;
+
+  /**
+   * Represents the resource type.
+   */
+  public readonly resourceType = 'cronjob';
+
+  constructor(scope: Construct, id: string, props: CronJobProps) {
+    super(scope, id, {
+      restartPolicy: RestartPolicy.NEVER,
+      select: false,
+      ...props,
+    });
+
+    this.apiObject = new k8s.KubeCronJob(this, 'Resource', {
+      metadata: props.metadata,
+      spec: Lazy.any({ produce: () => this._toKube() }),
+    });
+
+    if (props.startingDeadline != undefined && props.startingDeadline.toSeconds() < 10) {
+      throw new Error(`The starting deadline cannot be less than 10 seconds since the Kubernetes CronJobController checks things every 10 seconds and hence the CronJob may not be scheduled. The value passed is: ${props.startingDeadline}`);
```

Comment:
Try using the same names as the API, so its clear what needs to be changed. 

```suggestion
      throw new Error(`The 'startingDeadline' property cannot be less than 10 seconds since the Kubernetes CronJobController checks things every 10 seconds and hence the CronJob may not be scheduled. The value passed is: ${props.startingDeadline}`);
```

Reply:
Will update this.